doc/source/dev/development_environment.rst

```diff
@@ -238,6 +238,23 @@ Most python builds do not include debug symbols and are built with compiler
 optimizations enabled. To get the best debugging experience using a debug build
 of Python is encouraged, see :ref:`advanced_debugging`.
 
+In terms of debugging, NumPy also needs to be built in a debug mode. You need to use
+``debug`` build type and disable optimizations to make sure ``-O0`` flag is used
+during object building. To generate source-level debug information within the build process run::
+
+    $ CFLAGS="-O0 -g" CXXFLAGS="-O0 -g" spin build --clean -- -Dbuildtype=debug -Ddisable-optimization=true
```

Comment:
I wouldn't explicitly include `CFLAGS` in this invocation since `buildtype` should ideally handle this, but I would add a prominent note that on anaconda you'll need to also set `CFLAGS`.

Reply:
Sure! I removed them from the command and included in the conda note.